core/src/test/java/org/apache/calcite/test/MaterializedViewRelOptRulesTest.java

```diff
@@ -0,0 +1,1149 @@
+            resultContains(""
+            + "EnumerableCalc(expr#0=[{inputs}], expr#1=[CAST($t0):JavaType(int) NOT NULL], "
+            + "expr#2=[1], expr#3=[CAST($t1):INTEGER NOT NULL], expr#4=[=($t2, $t3)], "
+            + "EXPR$0=[$t1], $condition=[$t4])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])"))
+        .ok();
+  }
+
+  @Test void testJoinMaterialization7() {
+    sql("select \"depts\".\"name\"\n"
+            + "from \"emps\"\n"
+            + "join \"depts\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")",
+        "select \"dependents\".\"empid\"\n"
+            + "from \"emps\"\n"
+            + "join \"depts\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")\n"
+            + "join \"dependents\" on (\"depts\".\"name\" = \"dependents\".\"name\")")
+        .withChecker(
+            resultContains(""
+            + "EnumerableCalc(expr#0..2=[{inputs}], empid=[$t1])\n"
+            + "  EnumerableHashJoin(condition=[=($0, $2)], joinType=[inner])\n"
+            + "    EnumerableCalc(expr#0=[{inputs}], expr#1=[CAST($t0):VARCHAR], name=[$t1])\n"
+            + "      EnumerableTableScan(table=[[hr, MV0]])\n"
+            + "    EnumerableCalc(expr#0..1=[{inputs}], expr#2=[CAST($t1):VARCHAR], empid=[$t0], name0=[$t2])\n"
+            + "      EnumerableTableScan(table=[[hr, dependents]])"))
+        .ok();
+  }
+
+  @Test void testJoinMaterialization8() {
+    sql("select \"depts\".\"name\"\n"
+            + "from \"emps\"\n"
+            + "join \"depts\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")",
+        "select \"dependents\".\"empid\"\n"
+            + "from \"depts\"\n"
+            + "join \"dependents\" on (\"depts\".\"name\" = \"dependents\".\"name\")\n"
+            + "join \"emps\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")")
+        .withChecker(
+            resultContains(""
+            + "EnumerableCalc(expr#0..4=[{inputs}], empid=[$t2])\n"
+            + "  EnumerableHashJoin(condition=[=($1, $4)], joinType=[inner])\n"
+            + "    EnumerableCalc(expr#0=[{inputs}], expr#1=[CAST($t0):VARCHAR], proj#0..1=[{exprs}])\n"
+            + "      EnumerableTableScan(table=[[hr, MV0]])\n"
+            + "    EnumerableCalc(expr#0..1=[{inputs}], expr#2=[CAST($t1):VARCHAR], proj#0..2=[{exprs}])\n"
+            + "      EnumerableTableScan(table=[[hr, dependents]])"))
+        .ok();
+  }
+
+  @Test void testJoinMaterialization9() {
+    sql("select \"depts\".\"name\"\n"
+            + "from \"emps\"\n"
+            + "join \"depts\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")",
+        "select \"dependents\".\"empid\"\n"
+            + "from \"depts\"\n"
+            + "join \"dependents\" on (\"depts\".\"name\" = \"dependents\".\"name\")\n"
+            + "join \"locations\" on (\"locations\".\"name\" = \"dependents\".\"name\")\n"
+            + "join \"emps\" on (\"emps\".\"deptno\" = \"depts\".\"deptno\")")
+        .ok();
+  }
+
+  @Test void testJoinMaterialization10() {
```

Comment:
You should keep the slow tag. Now the slow tag is lost for many test cases.

Reply:
Thanks @hsyuan 
In current change, integration test is kept in `MaterializationTest`, it checks optimized plan and query result. `MaterializedViewSubstitutionVisitorTest` and `MaterializedViewRelOptRulesTest` checks the materialized view matching only by optimized plan, thus I don't think the plan will run slowly and I only removed slow tag for them

Reply:
Cool, thanks for explanation!